packages/wallets/src/hooks/useInputATMFormatter.ts

```diff
@@ -1,30 +1,69 @@
-import { useCallback, useEffect, useMemo, useRef } from 'react';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { unFormatLocaleString } from '@deriv/utils';
 import useInputDecimalFormatter from './useInputDecimalFormatter';
 
 type TOptions = {
     fractionDigits?: number;
     locale?: Intl.LocalesArgument;
+    maxDigits?: number;
 };
 
-const useInputATMFormatter = (initial?: number, options?: TOptions) => {
+const separatorRegex = /[,.]/g; // locale-agnostic
+
+const useInputATMFormatter = (inputRef: React.RefObject<HTMLInputElement>, initial?: number, options?: TOptions) => {
+    const input = inputRef.current;
+
     const isPasting = useRef(false);
     const { onChange: onChangeDecimal, value } = useInputDecimalFormatter(undefined, options);
-    const { fractionDigits = 2, locale } = options || {};
+    const { fractionDigits = 2, locale, maxDigits = 14 } = options || {};
 
     const formattedValue = useMemo(
         () => `${Number(value).toLocaleString(locale, { minimumFractionDigits: fractionDigits })}`,
         [fractionDigits, locale, value]
     );
 
+    // maintain the input state to compare new values to it on every change
+    const [prevFormattedValue, setPrevFormattedValue] = useState<string>(formattedValue);
+    useEffect(() => {
+        setPrevFormattedValue(formattedValue);
+    }, [formattedValue]);
+
+    // keep the caret from jumping
+    const [caretNeedsRepositioning, setCaretNeedsRepositioning] = useState<boolean>(false);
+    const [caret, setCaret] = useState<number>();
+    useEffect(() => {
+        if (caret && caretNeedsRepositioning && input) {
```

Comment:
might be cool to start adding tests for complex files like that - there is a lot of edge cases here waiting to be broken by someone in future,